### PR TITLE
Also try to remove broken symlinks during uninstall

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomvanmele** [comment]
If a user install a bunch of packages from an environment, and then removes the environment without uninstalling the packages this can create problems when trying to install again later on.

The uninstall process now also removes all broken symlinks...

**Review by Licini — APPROVED** [review]
Looks good!

**Comment by tomvanmele** [comment]
the following will now also happen during the install procedure:

1. all installed packages that are not available in the current env will be removed (albeit quietly)
2. all installed packages that are available will be reinstalled

**Comment by tomvanmele** [comment]
not sure if this is what we mean to happen though.
however, it seems to me that packages from a different env should not be allowed to float around in a new install...

File: src/compas_rhino/install.py [hunk 1]
```diff
@@ -62,10 +64,20 @@ def install(version=None, packages=None):
     symlinks_to_uninstall = []
     exit_code = 0
 
+    for name in os.listdir(scripts_path):
+        if name.startswith('compas') and not name.endswith('.py'):
```
Comment (beverlylytle) [inline src/compas_rhino/install.py]: Would it make sense for this to also include all `installable_rhino_packages`?  In the case that `packages` is provided by the user, then something like `roslibpy` (which is a default installable package for `compas_fab`) would be missed by this check and also would not be added to `packages` by `_filter_installable_packages` to be removed and added again.

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+* Changed `compas_rhino.uninstall` to also remove broken symlinks if no specific packages are provided for installation.
+* Changed `compas_rhino.install` to also remove broken symlinks if no specific packages are provided for installation.
```
Comment (beverlylytle) [inline CHANGELOG.md]: It looks like broken symlinks are removed whether or not packages are specified on install.

File: src/compas_rhino/uninstall.py [hunk 3]
```diff
@@ -45,13 +45,25 @@ def uninstall(version=None, packages=None):
...
+    ipylib_path = compas_rhino._get_ironpython_lib_path(version)
+
+    # If no specific packages are provided for uninstall
+    # everything should be removed,
+    # and not just the default packages
+    # or the packages that the bootstrapper is aware of.
+    if not packages:
+        packages = []
+        for name in os.listdir(scripts_path):
+            if name.startswith('compas') and not name.endswith('.py'):
```
Comment (beverlylytle) [inline src/compas_rhino/uninstall.py]: again, `roslibpy` or other default compas-installed packages?

File: CHANGELOG.md [hunk 4]
```diff
@@ -20,6 +20,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+* Changed `compas_rhino.uninstall` to also remove broken symlinks if no specific packages are provided for installation.
```
Comment (beverlylytle) [inline CHANGELOG.md]: 'for un-installation'?

File: src/compas_rhino/uninstall.py (hunk 3 above)
Comment (tomvanmele) [reply src/compas_rhino/uninstall.py]: is this also the case here?

Comment (beverlylytle) [reply src/compas_rhino/uninstall.py]: yes, i think so.

File: src/compas_rhino/install.py [hunk 5]
```diff
@@ -47,25 +47,55 @@ def install(version=None, packages=None):
...
     results = []
     symlinks_to_install = []
     symlinks_to_uninstall = []
     exit_code = 0
 
+    # check all installable packages
+    # add the packages that can't be imported from the current env to the list of symlinks to uninstall
+    # and remove the package name from the list of installable packages
+    # make a copy of the list to avoid problems with removing items
+    # note: perhaps this should already happen in the filter function...
```
Comment (beverlylytle) [inline src/compas_rhino/install.py:72]: This makes sense to me.

**Pull request merged**